Fails with db dir creation

06:27:28 161796:	Trying to geocode "Россия, г. Ставрополь, Северо-Западный"...
06:27:29 161796:	Successfully geocoded "Россия, г. Ставрополь, Северо-Западный" as lat:45.066298 lon:41.935898
06:27:29 161796:	Sending snapshot to pipeline...
06:27:29 161796:	Successfully notified the pipeline
06:27:29 161796:	Dumping card to disk...
06:27:29 161796:	Failed to create card dir: /db
06:27:29 161796:	Panic during fetching of card 161796:	Failed to create card dir: /db
panic: 161796:	Failed to create card dir: /db [recovered]
	panic: 161796:	Failed to create card dir: /db
goroutine 26 [running]:
github.com/LostPetInitiative/poiskzoo-ru-crawler/pkg/crawler.DoCardJob.func1()
	/app/pkg/crawler/downloader.go:25 +0xac
panic({0x6b7660, 0xc0002144c0})
	/usr/local/go/src/runtime/panic.go:884 +0x212
log.Panicf({0x720c79?, 0x3?}, {0xc0004b9df8?, 0x1?, 0x1?})
	/usr/local/go/src/log/log.go:395 +0x67

--- a/main.go
+++ b/main.go
@@ -93,6 +93,7 @@ func main() {
 			if err != nil {
 				log.Panic(err)
 			}
+			cardDirContent = make([]fs.DirEntry, 0)
 		} else {
 			log.Panic(err)
 		}

--- a/pkg/crawler/downloader.go
+++ b/pkg/crawler/downloader.go
@@ -1,7 +1,9 @@
 package crawler
 
 import (
+	"errors"
 	"fmt"
+	"io/fs"
 	"log"
 	"net/url"
 	"os"
@@ -26,6 +28,13 @@ func DoCardJob(card types.CardID, cardsDir string, notificationUrl *url.URL) {
 		}
 	}
 	defer cardJobFailurePrinter()
+
+	cardDir := path.Join(cardsDir, fmt.Sprintf("%d", card))
+	// workaround for paid promotion
+	// TODO: do something smarter
+	if _, err := os.Stat(cardDir); err == nil || !errors.Is(err, fs.ErrNotExist) {
+		log.Printf("%d:\t Card dir already exists. Consider it as processed. skipping it\n", card)
+	}
 
 	log.Printf("%d:\tFetching card...\n", card)
 	fetchedCard, err := GetPetCard(card)
@@ -92,9 +101,8 @@ func DoCardJob(card types.CardID, cardsDir string, notificationUrl *url.URL) {
 
 	log.Printf("%d:\tDumping card to disk...\n", card)
 
-	cardDir := path.Join(cardsDir, fmt.Sprintf("%d", card))
 	err = os.Mkdir(cardDir, 0644)
-	if err != nil {
+	if err != nil && !errors.Is(err, fs.ErrExist) {
 		log.Panicf("%d:\tFailed to create card dir: %v", card, cardsDir)
 	}
 

